### DISCUSSION TITLE
Query internal DB

### DISCUSSION THREAD

**Discussion by KernOil** [Budibase/budibase#1657] [Help]
Is there a way to query the internal DB? if so how can I view the internal DB externally / use it with BI tools. If I can get it into a data warehouse that should be good enough.

**Comment by mjashanks** [comment]
**EDIT: This is now documented at https://docs.budibase.com/self-hosting/accessing-couchdb**


Hey @AndrewJRamay.

Our internal tables use [CouchDB](https://couchdb.apache.org/). if you run `docker ps`, you should see an image with the name `ibm/couchdb3`. As you can see below, mine is running on port 4005

![image](https://user-images.githubusercontent.com/3524181/121180339-a5d7e980-c858-11eb-92b0-50456a3552fd.png)

If I then browse to `http://localhost:4005/_utils`, I see the interface below. You might need to expose the port, if accessing over a network.

![image](https://user-images.githubusercontent.com/3524181/121180721-10892500-c859-11eb-8131-0d93cf67df19.png)

This is CouchDB's management interface - Fauxton. You can find credentials for this inside your `.env` file, which lives in the same directory as your `docker-compose.yaml` file. Have a look for the lines:

```
COUCH_DB_PASSWORD=<your password>
COUCH_DB_USER=<your username>
```

Your app's db should be called something like `app_<appid>` - let me know if you need help determining the correct App ID.

You will see all of your data in here, but not necessarily in the most queryable format - depending on what you are used to.

CouchDB may have a direct connector to the data warehouse that you use - happy to assist further if you have any questions on what to do next.

**Reply by KernOil** [reply]
Awesome thank you! I will give that a shot today.

**Comment by mjashanks** [comment]
@tobehn - you asked about this.

You can skip the bit about `docker ps` and access CouchDb directly via `http://<domain>:10000/db/_utils/#` .

**Reply by tobehn** [reply]
Great, thank you @mjashanks.
